sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileScanRDD.scala

```diff
@@ -135,58 +135,44 @@ class FileScanRDD(
        * Only update metadata row when `currentFile` is changed.
        */
       private def updateMetadataRow(): Unit = {
-        if (metadataColumns.nonEmpty && currentFile != null) {
-          val path = new Path(currentFile.filePath)
-          metadataColumns.zipWithIndex.foreach { case (attr, i) =>
-            attr.name match {
-              case FILE_PATH => metadataRow.update(i, UTF8String.fromString(path.toString))
-              case FILE_NAME => metadataRow.update(i, UTF8String.fromString(path.getName))
-              case FILE_SIZE => metadataRow.update(i, currentFile.fileSize)
-              case FILE_MODIFICATION_TIME =>
-                // the modificationTime from the file is in millisecond,
-                // while internally, the TimestampType is stored in microsecond
-                metadataRow.update(i, currentFile.modificationTime * 1000L)
-            }
+        if (metadataColumns.isEmpty || currentFile == null) return
+        val path = new Path(currentFile.filePath)
+        metadataColumns.zipWithIndex.foreach { case (attr, i) =>
+          attr.name match {
+            case FILE_PATH => metadataRow.update(i, UTF8String.fromString(path.toString))
+            case FILE_NAME => metadataRow.update(i, UTF8String.fromString(path.getName))
+            case FILE_SIZE => metadataRow.update(i, currentFile.fileSize)
+            case FILE_MODIFICATION_TIME =>
+              // the modificationTime from the file is in millisecond,
+              // while internally, the TimestampType is stored in microsecond
+              metadataRow.update(i, currentFile.modificationTime * 1000L)
```

Comment:
nit: unnecessary change? didn't feel the readability improved much after negating the if condition.